### PR TITLE
Add support for connection options

### PR DESCRIPTION
**Pull request by IvoMiller**
Adds support for Encrypt and TrustServerConnection in Jenkins plugin

#### Why?
Allows users to configure these connection options - helps with Microsoft.Data.SqlClient change in behaviour for TrustServerCertificate

#### Who worked with you?
Noone

#### Automated testing
- New unit tests for bamboo code

#### Manual testing
Validate the following Jenkins behaviour on sales demo machine: 
- [x] existing pipeline works with latest SCA PS
- [x] existing pipeline works with old SCA PS
- [x] pipeline with new options sets those options with latest SCA PS
- [x] pipeline with new options succeeds with warnings with old SCA PS

#### Release Notes and Documentation
Updated release notes

#### Risk Assessment
Fairly small. This has been well tested manually.

### CODE REVIEW

File: src/main/resources/PowerShell/SqlCi.ps1 [hunk 1]
```diff
@@ -443,20 +466,29 @@ function TestDatabase
 
     if ($licenseSerialKey)
     {
-        Register-DlmSerialNumber $licenseSerialKey
+        Register-SqlChangeAutomation $licenseSerialKey
+    }
+
+    $connectionOptions = @{
+    }
+    if (AreConnectionOptionsHandled($temporaryDatabaseEncryptConnection, $temporarTestSourceOptionyDatabaseTrustServerCertificate)){
```
Comment (Rixium) [inline src/main/resources/PowerShell/SqlCi.ps1]: I fixed this in my branch for the azure stuff, but you may as well fix it here instead.
"temporarTestSourceOptionyDatabaseTrustServerCertificate"

Comment (IvoMiller) [reply src/main/resources/PowerShell/SqlCi.ps1]: ah. Good spot. It will need fixing in both as this is a copy of that file in a different repo

Comment (IvoMiller) [reply src/main/resources/PowerShell/SqlCi.ps1]: I do really hate that ps doesn't blow up for that kind of error

File: src/main/resources/PowerShell/SqlCi.ps1 [hunk 2]
```diff
@@ -471,7 +471,7 @@ function TestDatabase
 
     $connectionOptions = @{
     }
-    if (AreConnectionOptionsHandled($temporaryDatabaseEncryptConnection, $temporarTestSourceOptionyDatabaseTrustServerCertificate)){
+    if (AreConnectionOptionsHandled($temporaryDatabaseEncryptConnection, $temporaryDatabaseTrustServerCertificate)){
         $connectionOptions += @{ 'Encrypt' = $temporaryDatabaseEncryptConnection.ToBool() }
```
Comment (Rixium) [inline src/main/resources/PowerShell/SqlCi.ps1:475]: Is the ToBool necessary here, or should the above arguments also be converted. Not sure if AreConnectionOptionsHandled would work correctly if they're not bool going in.

`AreConnectionOptionsHandled($temporaryDatabaseEncryptConnection.ToBool(), $temporaryDatabaseTrustServerCertificate.ToBool())`

Comment (IvoMiller) [reply src/main/resources/PowerShell/SqlCi.ps1:475]: They are of type switch which is a bit weird I think. ToBool() was needed there to be converted to C# bool in Cmdlet code

**Review by Rixium — APPROVED** [review]
Looks good

**Pull request merged**